veles/secrets/postmanapikey/validator.go

```diff
@@ -0,0 +1,237 @@
+	}
+}
+
+// NewCollectionValidator creates a new ValidatorCollection with the given options.
+func NewCollectionValidator(opts ...ValidatorOptionCollection) *ValidatorCollection {
+	v := &ValidatorCollection{
+		httpC: http.DefaultClient,
+	}
+	for _, opt := range opts {
+		opt(v)
+	}
+	return v
+}
+
+// collectionErrorResponse models Postman's collection endpoint error JSON.
+type collectionErrorResponse struct {
+	Error struct {
+		Name    string `json:"name"`
+		Message string `json:"message"`
+	} `json:"error"`
+}
+
+const (
+	// Exact values observed for a valid-but-not-authorized response.
+	forbiddenErrorName    = "forbiddenError"
+	forbiddenErrorMessage = "You are not authorized to perform this action."
+)
+
+// Validate checks whether the given PostmanCollectionToken is valid.
+//
+// It calls GET {collectionEndpoint}?access_key=<token> using a dummy
+// collection ID. The dummy collection ID is used to create a predictable
+// authorization failure for valid tokens (403 Forbidden) while invalid
+// tokens produce 401 Authentication errors.
+//
+// Interpretation of statuses:
+//   - 200 OK  -> token is valid and authorized for the collection (rare here).
+//   - 403     -> authenticated but not authorized for this collection -> valid
+//     *only if* the response body matches the exact JSON produced by
+//     Postman for that situation:
+//     {"error":{"name":"forbiddenError","message":"You are not authorized to perform this action."}}
+//   - 401     -> invalid token (authentication failure).
+//   - other   -> validation failed.
+func (v *ValidatorCollection) Validate(ctx context.Context, key PostmanCollectionToken) (veles.ValidationStatus, error) {
+	url := collectionEndpoint + "?access_key=" + key.Key
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return veles.ValidationFailed, fmt.Errorf("unable to create HTTP request: %w", err)
+	}
+
+	res, err := v.httpC.Do(req)
+	if err != nil {
+		return veles.ValidationFailed, fmt.Errorf("unable to GET %q: %w", url, err)
+	}
+	defer res.Body.Close()
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		// 200 => token is valid and has access to the collection.
+		return veles.ValidationValid, nil
+
+	case http.StatusForbidden:
+		// 403 => usually means the token authenticated but is not authorized
+		// for this collection. Because we intentionally used a collection ID
+		// we don't own, this indicates the token itself may be valid.
+		// To avoid false positives we require the response body to match the
+		// exact JSON Postman returns for that case (name + message).
+		var resp collectionErrorResponse
+		if err := json.NewDecoder(res.Body).Decode(&resp); err != nil {
+			// Decoding failed -> ambiguous response, treat as failed to validate.
+			return veles.ValidationFailed, fmt.Errorf("unable to parse response from %q: %w", url, err)
+		}
+		if resp.Error.Name == forbiddenErrorName && resp.Error.Message == forbiddenErrorMessage {
```

Comment:
I'd say we should just check for the name, not the error message.

APIs are subject to change and while the HTTP status code and the name could be more stable the detailed message might update more often, so this validation method might go stale over time.